src/anonymous-credentials/types-and-consts.ts

```diff
@@ -57,3 +83,40 @@ export const ACCUMULATOR_PROVING_KEY_LABEL_BYTES = te.encode(ACCUMULATOR_PROVING
 // Label used for generating SAVER encryption generators
 export const SAVER_ENCRYPTION_GENS_LABEL = 'DockSAVEREncryptionGens2022';
 export const SAVER_ENCRYPTION_GENS_BYTES = te.encode(SAVER_ENCRYPTION_GENS_LABEL);
+
+function once<T>(f: () => T) {
+  const NO_VALUE = {};
+  let calculated: typeof NO_VALUE | T = NO_VALUE;
+
+  return function () {
+    if (calculated === NO_VALUE) {
+      return (calculated = f());
+    } else {
+      return calculated;
+    }
+  };
+}
+
+export const DEFAULT_SIGNATURE_LABEL_BYTES = Object.setPrototypeOf(
+  {
+    [BBSSignatureParams.name]: BBS_SIGNATURE_PARAMS_LABEL_BYTES,
+    [BBSPlusSignatureParamsG1.name]: BBS_PLUS_SIGNATURE_PARAMS_LABEL_BYTES,
+    [PSSignatureParams.name]: PS_SIGNATURE_PARAMS_LABEL_BYTES
+  },
+  null
+);
+
+export const DEFAULT_SIGNATURE_PARAMS = Object.setPrototypeOf(
+  {
+    [BBSSignatureParams.name]: once(() =>
+      BBSSignatureParams.generate(2, DEFAULT_SIGNATURE_LABEL_BYTES[BBSSignatureParams.name])
```

Comment:
This is always going to be wasteful as all creds have more than 2 attributes

Reply:
I don't understand your point because the same thing happens at the current code https://github.com/docknetwork/crypto-wasm-ts/blob/master/src/anonymous-credentials/presentation-builder.ts#L291.
The current code is better because these parameters will be initialized once while the previous code recreated them every time.
Although we may define another amount of parameters.

Reply:
The existing code that you referenced is wastefull as well. `sigParams` (of corresponding scheme) should be unset at the beginning and should be set the first time it needs to be used and should be created with `numAttribs` (in that context)